### PR TITLE
Add easy way to select current month as range filter

### PR DESCRIPTION
**Pull request by CoffeeIsLife87**
Trying to stay on top of my personally finance has made me aware of what is probably my only complaint about this software. The complaint being that there is no easy way to filter by month. 

Currently filtering to a specific month involves 1. setting the start and end year if it isn't correct to begin with, 2. setting both start and end months, and 3. setting the end date to the last of the month. I'm glad that's an option but it's a lot of extra clicks that I believe could be reduced in a clean manner. What this PR does is adds 1 button above the calendar preview that will fill in the range select boxes with whatever month is currently shown in said preview.

This PR is marked as draft since what I threw together isn't quite perfectly polished. I've never written in C# but I did compile and run (Done on Fedora 38 with gnome builder) to make sure it worked properly. However if the code needs to be cleaned up I'd be happy to look into doing that. As for the UI, I wasn't sure if there was a different icon that I should have used. The tooltip is also open to suggestions if you think it could be phrased better

### CODE REVIEW

**Comment by fsobolev** [comment]
Thanks for your contribution! There are some minor things to change, and I think the icon should be different, but overall the code looks good to me 😊 

image

`month-symbolic` icon looks similar to `today-symbolic`, but it perfectly fits the purpose and I think it's still clear that this is a separate button.

File: NickvisionMoney.GNOME/Blueprints/account_view.blp [hunk 1]
```diff
@@ -175,16 +175,27 @@ Adw.Bin _root {
...
-            tooltip-text: _("Reset To Today");
+          header-suffix: Gtk.Box 
+          {
+            orientation: horizontal;
 
-            Adw.ButtonContent {
-              icon-name: "today-symbolic";
-              label: _("Today");
+            Gtk.Button _selectWholeMonthButton {
+              tooltip-text: _("Select current month");
```
Comment (fsobolev) [inline NickvisionMoney.GNOME/Blueprints/account_view.blp:183]:
```suggestion
              tooltip-text: _("Select Current Month");
```

To match other tooltip labels.

File: NickvisionMoney.GNOME/Views/AccountView.cs [hunk 2]
```diff
@@ -1211,6 +1214,36 @@ private void OnCalendarSelectedDateChanged(Gtk.Calendar sender, EventArgs e)
         }
     }
 
+    /// <summary>
+    /// occurs when calendar select all button is clicked
+    /// </summary>
+    /// <param name="sender">Gtk.Button</param>
+    /// <param name="e">EventArgs</param>
+    private void OnSelectCurrentTime(Gtk.Button sender, EventArgs e)
```
Comment (fsobolev) [inline NickvisionMoney.GNOME/Views/AccountView.cs:1222]: `OnSelectCurrentMonth` maybe?

File: NickvisionMoney.GNOME/Views/AccountView.cs [hunk 3]
```diff
@@ -1211,6 +1214,36 @@ private void OnCalendarSelectedDateChanged(Gtk.Calendar sender, EventArgs e)
         }
     }
 
+    /// <summary>
+    /// occurs when calendar select all button is clicked
```
Comment (fsobolev) [inline NickvisionMoney.GNOME/Views/AccountView.cs:1218]:
```suggestion
    /// Occurs when the select current month button is clicked
```

File: NickvisionMoney.GNOME/Views/AccountView.cs [hunk 4]
```diff
@@ -1211,6 +1214,36 @@ private void OnCalendarSelectedDateChanged(Gtk.Calendar sender, EventArgs e)
...
+    private void OnSelectCurrentTime(Gtk.Button sender, EventArgs e)
+    {
+        _rangeExpander.SetEnableExpansion(true);
+        var selectedDay = gtk_calendar_get_date(_calendar.Handle);
+        var selectedMonth = (uint)(g_date_time_get_month(ref selectedDay) - 1);
+        var selectedYear = g_date_time_get_year(ref selectedDay);
+        var yearsForRangeFilter = _controller.YearsForRangeFilter.ToArray();
+
+        uint selectedYearIndex = 0;
+        for (uint yearPosition = 0; yearPosition < yearsForRangeFilter.Length; yearPosition++) {
```
Comment (fsobolev) [inline NickvisionMoney.GNOME/Views/AccountView.cs:1231]:
```suggestion
        var selectedYearIndex = 0u;
        for (var yearPosition = 0u; yearPosition < yearsForRangeFilter.Length; yearPosition++) {
```

File: NickvisionMoney.GNOME/Views/AccountView.cs [hunk 5]
```diff
@@ -1211,6 +1214,36 @@ private void OnCalendarSelectedDateChanged(Gtk.Calendar sender, EventArgs e)
...
+    /// <param name="e">EventArgs</param>
+    private void OnSelectCurrentTime(Gtk.Button sender, EventArgs e)
+    {
+        _rangeExpander.SetEnableExpansion(true);
+        var selectedDay = gtk_calendar_get_date(_calendar.Handle);
+        var selectedMonth = (uint)(g_date_time_get_month(ref selectedDay) - 1);
+        var selectedYear = g_date_time_get_year(ref selectedDay);
+        var yearsForRangeFilter = _controller.YearsForRangeFilter.ToArray();
+
+        uint selectedYearIndex = 0;
```
Comment (fsobolev) [inline NickvisionMoney.GNOME/Views/AccountView.cs:1230]:
```suggestion
        var yearsForRangeFilter = _controller.YearsForRangeFilter.ToArray();
        uint selectedYearIndex = 0;
```

File: NickvisionMoney.GNOME/Views/AccountView.cs [hunk 6]
```diff
@@ -1211,6 +1214,36 @@ private void OnCalendarSelectedDateChanged(Gtk.Calendar sender, EventArgs e)
...
+                break;
+            }
+        }
+        _startYearDropDown.SetSelected(selectedYearIndex);
+        _endYearDropDown.SetSelected(selectedYearIndex);
+        
+        _startMonthDropDown.SetSelected(selectedMonth);
+        _endMonthDropDown.SetSelected(selectedMonth);
+        
+        _startDayDropDown.SetSelected(0);
```
Comment (fsobolev) [inline NickvisionMoney.GNOME/Views/AccountView.cs:1243]:
```suggestion
        _endYearDropDown.SetSelected(selectedYearIndex);
        _startMonthDropDown.SetSelected(selectedMonth);
        _endMonthDropDown.SetSelected(selectedMonth);
        _startDayDropDown.SetSelected(0);
```

**Review by fsobolev — CHANGES_REQUESTED** [review]

File: NickvisionMoney.GNOME/Blueprints/account_view.blp (hunk 1 above)
Comment (CoffeeIsLife87) [reply NickvisionMoney.GNOME/Blueprints/account_view.blp]: I added everything and made sure it built before committing!! Should be good to go

**Review by fsobolev — APPROVED** [review]

File: NickvisionMoney.GNOME/Resources/org.nickvision.money.gresource.xml [hunk 7]
```diff
@@ -18,6 +18,7 @@
     <file preprocess="xml-stripblanks">resources-symbolic.svg</file>
     <file preprocess="xml-stripblanks">sun-outline-symbolic.svg</file>
     <file preprocess="xml-stripblanks">today-symbolic.svg</file>
+    <file preprocess="xml-stripblanks">month-symbolic.svg</file>
```
Comment (nlogozzo) [inline NickvisionMoney.GNOME/Resources/org.nickvision.money.gresource.xml:21]: Can you sort this alphabetically in the list?

**Review by nlogozzo — CHANGES_REQUESTED** [review]

**Review by nlogozzo — CHANGES_REQUESTED** [review]

**Comment by nlogozzo** [comment]
Thank you for the contribution! ❤️

**Review by nlogozzo — APPROVED** [review]

**Comment by nlogozzo** [comment]
https://github.com/NickvisionApps/Denaro/releases/tag/2023.8.0-beta1

**Pull request merged**